backend/src/target.ts

```diff
@@ -76,6 +77,43 @@ class Target {
     return subscription;
   }
 
+  sendBlocksBatchTx(
+    feedId: U64,
+    signer: SignerWithAddress,
+    txData: BatchTxBlock[],
+    nonce?: bigint,
+  ): Promise<Hash> {
+    this.logger.debug(`Sending ${txData.length} blocks to feed: ${feedId}`);
+    this.logger.debug(`Signer: ${signer.address}`);
+
+    const putCalls = txData.map(({ block, metadata }: BatchTxBlock) => {
+      return this.api.tx.feeds.put(feedId, u8aToHex(block), u8aToHex(metadata));
+    });
+    return new Promise((resolve, reject) => {
+      let unsub: () => void;
+      this.api.tx.utility
+        .batchAll(putCalls)
+        // it is required to specify nonce, otherwise transaction within same block will be rejected
+        // if nonce is -1 API will do the lookup for the right value
+        // https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+        .signAndSend(signer.address, { nonce: nonce || -1, signer }, (result) => {
+          if (result.isError) {
+            reject(result.status.toString());
+            unsub();
+          } else if (result.status.isInBlock) {
+            resolve(result.status.asInBlock);
```

Comment:
probably it will be more useful to resolve block hash

Reply:
This is a block hash, exactly the same code as in the regular version above